src/datasource.tsx

```diff
@@ -554,9 +554,10 @@ export class PrometheusDatasource
 
     // Align query interval with step to allow query caching and to ensure
     // that about-same-time query results look the same.
-    const adjusted = alignRange(start, end, query.step, this.timeSrv.timeRange().to.utcOffset() * 60);
-    query.start = adjusted.start;
-    query.end = adjusted.end;
+    // Removed alignRange to avoid calendar boundary issues.
+    // const adjusted = alignRange(start, end, query.step, this.timeSrv.timeRange().to.utcOffset() * 60);
```

Comment:
Are there cases when this function was useful?
If we don't need it, should its code and tests to be removed as well?

Reply:
I don’t think so. I’ve removed the function and the related tests.